lib/engine/game/g_1812/step/buy_train.rb

```diff
@@ -0,0 +1,87 @@
+# frozen_string_literal: true
+
+require_relative '../../../step/base'
+require_relative '../../../step/buy_train'
+require_relative '../../../step/automatic_loan'
+
+module Engine
+  module Game
+    module G1812
+      module Step
+        class BuyTrain < Engine::Step::BuyTrain
```

Comment:
Could this class inherit from `G1867::Step::BuyTrain`? You've got several methods that are cut'n'pasted from that class.

Reply:
100% correct. I wasn't sure how to inherit a couple of methods directly from Engine::Step::BuyTrain, but I think I got it now. I needed to inherit from Engine because the 1867 methods included code I didn't want, which prevented me from calling `super`. Let me know if I can improve this, it's in update `fixes buy_train.rb inheritance`